torch/distributed/distributed_c10d.py

```diff
@@ -4115,3 +4116,35 @@ def _get_group_tag(pg: ProcessGroup) -> str:
     if tag.startswith("user:"):
         tag = tag[5:]
     return tag
+
+
+# This ops are not friently to TorchDynamo. So, we decide to disallow these ops
+# in FX graph, allowing them to run them on eager, with torch.compile.
+dynamo_unsupported_distributed_c10d_ops = [
```

Comment:
I am missing the context here. Is the idea just to exclude all the operations defined in `distributed_c10d.py`?

Reply:
Yes

Reply:
Or specifically, the operators that do not work well with TorchDynamo tracing and we want to graph break (i.e. fallback to eager) on them

Reply:
@H-Huang ping

Reply:
I see, then would it be possible to just get all operations of the module through something like:
https://stackoverflow.com/questions/139180/how-to-list-all-functions-in-a-module

Then we don't have to create this list in `distributed_c10d` and worry about keeping it up to date

Reply:
Yes, so we tried that but then it gives a large number of functions and many of them not really "operators".

Another way I tried earlier in a commit - https://github.com/pytorch/pytorch/pull/100071/commits/8e6dee36ff3eea0648222ca739458ebf140cb4a1 
was to record the ops in exception_handler. The assumption was that anyone adding a new op will decorate it withe exception handler. But, that looked little hacky as well.

Reply:
Thanks for pointing that out. Removed it from the list.